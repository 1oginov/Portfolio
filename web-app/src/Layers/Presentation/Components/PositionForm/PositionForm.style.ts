import { withStyles } from '@material-ui/core';

export const style = withStyles((theme) => ({
  buttons: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  root: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
  },
  row: {
    marginBottom: theme.spacing(1),
    marginTop: theme.spacing(1),
    textAlign: 'center',
  },
}));
