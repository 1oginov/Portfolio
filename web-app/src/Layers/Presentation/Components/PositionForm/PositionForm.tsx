import classNames from 'classnames';
import { Button, TextField } from '@material-ui/core';
import * as React from 'react';

import {
  PositionFormEnhancerInputProps, PositionFormEnhancerProps,
} from 'Layers/Behavior/Enhancers/PositionFormEnhancer/PositionFormEnhancer';

export interface PositionFormProps extends PositionFormEnhancerInputProps {
  backButton?: React.ReactNode;
  classes: { [key: string]: string };
}

type Props = PositionFormProps & PositionFormEnhancerProps;

export const PositionForm: React.FunctionComponent<Props> = ({
  amount, backButton, classes, closeCommission, closeDate, closePrice, handleAmountChange, handleCloseCommissionChange,
  handleCloseDateChange, handleClosePriceChange, handleOpenCommissionChange, handleOpenDateChange,
  handleOpenPriceChange, handleSubmit, handleSymbolChange, openCommission, openDate, openPrice, symbol,
}: Props) => (
  <form className={classes.root} onSubmit={handleSubmit}>

    <div className={classes.row}>
      <TextField
        fullWidth
        label="Symbol"
        onChange={handleSymbolChange}
        required
        type="text"
        value={symbol}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        inputProps={{ min: 1 }}
        label="Amount"
        onChange={handleAmountChange}
        required
        type="number"
        value={amount}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        inputProps={{ min: 0, step: 0.01 }}
        label="Open Price"
        onChange={handleOpenPriceChange}
        required
        type="number"
        value={openPrice}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        inputProps={{ min: 0, step: 0.01 }}
        label="Open Commission"
        onChange={handleOpenCommissionChange}
        required
        type="number"
        value={openCommission}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        label="Open Date"
        onChange={handleOpenDateChange}
        required
        type="date"
        value={openDate}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        inputProps={{ min: 0, step: 0.01 }}
        label="Close Price"
        onChange={handleClosePriceChange}
        type="number"
        value={closePrice}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        inputProps={{ min: 0, step: 0.01 }}
        label="Close Commission"
        onChange={handleCloseCommissionChange}
        type="number"
        value={closeCommission}
        variant="outlined"
      />
    </div>

    <div className={classes.row}>
      <TextField
        fullWidth
        label="Close Date"
        onChange={handleCloseDateChange}
        type="date"
        value={closeDate}
        variant="outlined"
      />
    </div>

    <div className={classNames(classes.row, backButton && classes.buttons)}>
      {backButton}
      <Button color="primary" variant="contained" type="submit">Update</Button>
    </div>

  </form>
);
