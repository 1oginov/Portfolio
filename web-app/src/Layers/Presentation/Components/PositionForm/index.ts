import { PositionFormEnhancer } from 'Layers/Behavior/Enhancers/PositionFormEnhancer/PositionFormEnhancer';

import { PositionForm as PositionFormComponent, PositionFormProps } from './PositionForm';
import { style } from './PositionForm.style';

export const PositionForm = style(PositionFormEnhancer<PositionFormProps>()(PositionFormComponent));
