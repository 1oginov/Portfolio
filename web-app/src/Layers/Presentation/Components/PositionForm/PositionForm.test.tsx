/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import classes from 'identity-obj-proxy';
import * as React from 'react';

import { PositionForm } from './PositionForm';

it('matches snapshot', () => {
  const wrapper = shallow(
    <PositionForm
      amount={1}
      classes={classes}
      closeCommission={1}
      closeDate="2019-07-25"
      closePrice={100}
      displayedFields={[
        'symbol', 'amount', 'openPrice', 'openCommission', 'openDate', 'closePrice', 'closeCommission', 'closeDate',
      ]}
      handleAmountChange={() => undefined}
      handleCloseCommissionChange={() => undefined}
      handleCloseDateChange={() => undefined}
      handleClosePriceChange={() => undefined}
      handleOpenCommissionChange={() => undefined}
      handleOpenDateChange={() => undefined}
      handleOpenPriceChange={() => undefined}
      handleSubmit={() => undefined}
      handleSymbolChange={() => undefined}
      initialData={{
        amount: 1,
        closeCommission: 1,
        closeDate: '2019-07-25',
        closePrice: 100,
        openCommission: 1,
        openDate: '2019-07-02',
        openPrice: 100,
        symbol: 'AAPL',
      }}
      onSubmit={() => undefined}
      openCommission={1}
      openDate="2019-07-02"
      openPrice={100}
      requiredFields={[
        'symbol', 'amount', 'openPrice', 'openCommission', 'openDate',
      ]}
      submitButtonTitle="Submit"
      symbol="AAPL"
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
