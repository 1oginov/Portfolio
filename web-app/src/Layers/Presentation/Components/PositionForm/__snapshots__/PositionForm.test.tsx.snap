// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="root"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Symbol"
      onChange={[Function]}
      required={true}
      type="text"
      value="AAPL"
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      inputProps={
        Object {
          "min": 1,
        }
      }
      label="Amount"
      onChange={[Function]}
      required={true}
      type="number"
      value={1}
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      inputProps={
        Object {
          "min": 0,
          "step": 0.01,
        }
      }
      label="Open Price"
      onChange={[Function]}
      required={true}
      type="number"
      value={100}
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      inputProps={
        Object {
          "min": 0,
          "step": 0.01,
        }
      }
      label="Open Commission"
      onChange={[Function]}
      required={true}
      type="number"
      value={1}
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Open Date"
      onChange={[Function]}
      required={true}
      type="date"
      value="2019-07-02"
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      inputProps={
        Object {
          "min": 0,
          "step": 0.01,
        }
      }
      label="Close Price"
      onChange={[Function]}
      type="number"
      value={100}
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      inputProps={
        Object {
          "min": 0,
          "step": 0.01,
        }
      }
      label="Close Commission"
      onChange={[Function]}
      type="number"
      value={1}
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Close Date"
      onChange={[Function]}
      type="date"
      value="2019-07-25"
      variant="outlined"
    />
  </div>
  <div
    className="row"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      type="submit"
      variant="contained"
    >
      Update
    </WithStyles(ForwardRef(Button))>
  </div>
</form>
`;
